### DISCUSSION TITLE
Support setting language for toot, and respect the set language when rendering toots in timelines

### DISCUSSION THREAD

**Discussion by MarcoZehe** [NickColley/semaphore#62] [New features and ideas]
Right now, toots default to the language that is either set by the Mastodon instance language of the user, or the preferred language set by the user in their profile on Mastodon. However, for each toot, the language can be changed in the Mastodon web client as well as many others, like Mona or Toot!App on iOS. Likewise, if a toot is in a particular language, the toot gets rendered with a language attribute set appropriately, so screen readers will switch to the particular synthesizer language if supported.

Semaphore should do the same. While it is appropriate to set the main language on the whole page level to the language Semaphore is licalized in, by default U.S. English, if a toot has a different language, like German, right now the synthesizer will pronounce that toot in English instead, making it literally unintelligible. The separate articles for each toot should get a lang attribute override for the specified language of a toot.

Likewise, the composer should be enhanced with a dropdown that gives a language selection, defaulting to the language set either in the profile or by the Mastodon instance language of the user. That way, if I compose a toot in a different language, I can tell Mastodon and all other clients that, and they can respect that.

**Comment by NickColley** [comment]
This is important to resolve given that the wrong language results in unintelligible announcements, marked as accepted.

**Comment by mirabilos** [comment]
FediText has a preference setting in which the user can choose multiple languages to fill the language selector dropdown in the compose box with. This way, you can choose at compose time between your handful of favourite languages (including “no linguistic context” (e.g. just a smiley or a name) and “unknown language”) instead of the full list.

I kinda like this and would love for Semaphore, the only other client I use, to have that.
